From e47f5e728cef7a1b4032f590942f853e49a28121 Mon Sep 17 00:00:00 2001
From: "chungho.shin" <chungho.shin@coasianexell.com>
Date: Mon, 10 Aug 2020 15:28:42 +0900
Subject: [PATCH 1/1] [Yocto s5p6818] OPTEE Build Makefile modification

This patch is for the OPTEE Build Makefile modification.

When building Android Pie, the build structure was changed from
the existing script method to the lunch & make method.
There was a history of changing the path of the optee Make File
by multi-thread build.
so when Yocto Build, you have to apply this fix.

Signed-off-by: chungho.shin <chungho.shin@coasianexell.com>
Change-Id: I0bbe96ffd313ad1a8b27766783e67853089dffe9
---
 Makefile | 15 +++++++++------
 1 file changed, 9 insertions(+), 6 deletions(-)

diff --git a/Makefile b/Makefile
index 305fe46..200aa91 100644
--- a/Makefile
+++ b/Makefile
@@ -9,6 +9,9 @@ PLAT_UART_BASE ?= 0xC00A3000
 PLAT_DRAM_SIZE ?= 1024
 BL31_ON_SRAM = 1
 
+YOCTO_DEFCONFIG =
+YOCTO_DTB =
+
 ifeq ($(V),1)
   Q :=
   ECHO := @:
@@ -128,8 +131,8 @@ build-bl33:: $(aarch64-linux-gnu-gcc)
 build-bl33 $(BL33)::
 	$(ECHO) '  BUILD   $@'
 	$(Q)set -e ; cd u-boot-artik7 ; \
-	    $(MAKE) artik710_raptor_config ; \
-	    $(MAKE) CROSS_COMPILE="$(CROSS_COMPILE)"
+	    $(MAKE) ${YOCTO_DEFCONFIG} ; \
+	    $(MAKE) CROSS_COMPILE="$(CROSS_COMPILE)" LDFLAGS="" CC="${CC}"
 	$(Q)touch ${BL33}
 
 clean-bl33:
@@ -157,7 +160,7 @@ BL32 = secos/out/kernel-install/secos.bin
 else
 BL32 = optee_os/out/arm-plat-s5p6818/core/tee.bin
 endif
-BL33 = ../u-boot/u-boot-2016.01/u-boot.bin
+#BL33 = ../u-boot/u-boot-2016.01/u-boot.bin
 FIP = $(ATF)/fip.bin
 FIPloader = $(ATF)/fip-loader.bin
 FIPsecure = $(ATF)/fip-secure.bin
@@ -303,7 +306,7 @@ clean-lloader:
 # each time it is run
 
 LINUX = linux/arch/arm64/boot/Image
-DTB = nexell/s5p6818-artik710-raptor.dtb
+DTB = nexell/${YOCTO_DTB}
 DTB2 = linux/arch/arm64/boot/dts/$(DTB)
 # Config fragments to merge with the default kernel configuration
 KCONFIGS += linux/arch/arm64/configs/s5p6818_drone_defconfig
@@ -413,9 +416,9 @@ endif
 
 .PHONY: build-singleimage
 build-singleimage:: $(singleimage-deps)
-build-singleimage:: optee_build/gen_singleimage.sh
+build-singleimage:: ./gen_singleimage.sh
 	$(ECHO) "  GEN    $(SINGLE_IMG)"
-	$(Q)optee_build/gen_singleimage.sh $(SINGLE_PARAM)
+	$(Q)./gen_singleimage.sh $(SINGLE_PARAM)
 
 clean-singleimage:
 	$(ECHO) "  CLEAN  $@"
-- 
2.17.1

